ddgr: 1.1 -> 1.6

ceedubs opened the pull request:
###### Motivation for this change

ddgr (DuckDuckGo CLI) 1.6 no longer requires the `requests` Python
package and includes the commit that patched zsh-completion.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

dotlambda commented on pkgs/applications/misc/ddgr/default.nix:
```diff
@@ -1,29 +1,17 @@
 {stdenv, fetchpatch, fetchFromGitHub, python3Packages}:
```
```suggestion
{stdenv, fetchpatch, fetchFromGitHub, python3}:
```

ceedubs replied:
I thought about making this change, but I thought that keeping the argument passed in as `python3Packages` might help avoid breakage of anyone manually passing arguments into this derivation. It's also possible that I'm mistaken. Happy to make the change if that's desired.

dotlambda commented on pkgs/applications/misc/ddgr/default.nix:
```diff
@@ -1,29 +1,17 @@
-     url = "https://github.com/jarun/ddgr/commit/10c1a911a3d5cbf3e96357c932b0211d3165c4b8.patch";
-    })
-  ];
+  buildInputs = [ python3Packages.python ];
```
```suggestion
  buildInputs = [ python3 ];
```

dotlambda reviewed (changes requested):
Please add yourself to `maintainer-list.nix` in a separate commit.

ceedubs committed:
Add Cody Allen (ceedubs) to maintainer list

ceedubs commented:
Thanks for the review, @dotlambda! Changes incoming shortly.

ceedubs committed:
ddgr: 1.1 -> 1.6

ddgr (DuckDuckGo CLI) 1.6 no longer requires the `requests` Python
package and includes the commit that patched zsh-completion.

ceedubs commented:
Review suggestions addressed.

dotlambda commented:
@GrahamcOfBorg build ddgr
